### DISCUSSION TITLE
why this history state does not work?

### DISCUSSION THREAD

**Discussion by deleite** [statelyai/xstate#3661] [Q&A] [answered]
Hi all, why does history does not send me back to the last state within Form State?

I looked it up but the same example in the docs is the example from the eggheadio video, which did not really help.

https://stately.ai/registry/editor/72c99bf2-9d65-488b-b64f-e90ac55d4dd5?machineId=7e87ee05-5311-44e7-8b78-d6b79557fc6c

**Comment by deleite** [comment]
closest I could find to this was:
https://github.com/statelyai/xstate/issues/188#issuecomment-819095230

But a lot of code just to have that. Maybe I am thinking about the history state wrong here?

**Comment by Andarist** [comment, marked as answer]
It would be helpful to know what sequence of events are you generating here and what exactly you expect and where the model doesn't match your expectations.

I think though that the given model won't be able to utilize history states. The problem here is that you have made your Confirm state part of the Form state. So when you reach your history state then it might be able to target the last known state configuration of its parent (Form) but that is... Confirm.

**Reply by deleite** [reply]
I see now. Thanks for the explanation. 
I needed to move the confirm state outside form state, take a look now.
Now it works as intended, which is, for example I am on Valid state and I do onClose (notPristine) I need to confirm and if I choose no, I am back on the same state within form state.

**Reply by Andarist** [reply]
LGTM :)
